adapters/http-vertx-base/src/main/java/org/eclipse/hono/adapter/http/AbstractVertxBasedHttpProtocolAdapter.java

```diff
@@ -675,7 +675,7 @@ private void doUploadMessage(
                         responseReady,
                         currentSpan));
 
-        CompositeFuture.all(senderTracker, commandConsumerTracker)
+        CompositeFuture.join(senderTracker, commandConsumerTracker)
```

Comment:
I think I understood what happens here. But I think it would be worth spending a few words, in a source code comment, of why "join" is needed here.

Aside from that I would wish we could split of those nested monster methods and make them easier to understand.

Reply:
I agree. IMHO adding the comment shouldn't be a big thing. Splitting up the *monster* might deserve a separate issue. Would that work for everybody?

Reply:
I agree about the comment as well and I've amended the commit accordingly.

And yes, I also would like to see the method refactored. IMHO, this should be done in a separate PR though.

Reply:
> Would that work for everybody?

Absolutely! It was just a general remark, after trying to understand what is going on.